packages/upward-js/lib/ResolverVisitor.js

```diff
@@ -5,7 +5,8 @@ const { zipObject } = require('lodash');
 const isPrimitive = require('./isPrimitive');
 
 class ResolverVisitor {
-    constructor(io, rootDefinition, context) {
+    constructor(upwardPath, io, rootDefinition, context) {
```

Comment:
I noticed we are modifying lots of input args. In general I think it's safer to append new arguments instead of prepend.

Also, some docs on this file would go a long way :D

Reply:
Yeah, I suppose this shouldn't be a prepend :/ I'll refactor it